src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Integer MIN_PASSWORD_LENGTH = 6;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty() || email == null || password == null
+                || email.isEmpty() || password.isEmpty()
+                || !HashUtil.hashPassword(password, userFromDbOptional.get().getSalt())
+                .equals(userFromDbOptional.get().getPassword())) {
+            throw new AuthenticationException("Can't login with input data. Email: " + email);
+        }
+        return userFromDbOptional.get();
```

Comment:
you use the get() method three times. Perhaps it is better to make the type User instead of Option<User>? This way you will get rid of the isEmpty check